app/javascript/packs/react-bootloader.jsx

```diff
@@ -20,5 +21,33 @@ export const initReact = (mappings) => {
         render(elem, generator(data))
       })
     }
+
+    document
+      .querySelectorAll('[data-tooltip-type][data-tooltip-url]')
+      .forEach((elem) => {
+        const name = elem.dataset['tooltipType'] + '-tooltip'
+        const generator = mappings[name]
+
+        const componentData = { endpoint: elem.dataset['tooltipUrl'] }
+        const component = generator(componentData)
+
+        // Create an element render the React component in
+        const tooltipElem = document.createElement('div')
+        elem.insertAdjacentElement('afterend', tooltipElem)
+
+        // Link the tooltip element with the reference element
+        const popperOptions = {
+          placement: elem.dataset['placement'] || 'auto',
+        }
+        createPopper(elem, tooltipElem, popperOptions)
+
+        elem.addEventListener('mouseenter', () =>
+          render(tooltipElem, component)
```

Comment:
This renders a new React component each time the `mouseenter` event is triggered. I don't know if this is horribly inefficient though. We create the accompanying HTML element once per element. We could consider having just one tooltip element available, as I don't think we'll have overlapping tooltips.

Reply:
It's probably fine. It should not only trigger on `mousenter`. You probably want to trigger it too when the element focusses via keyboard, or just focusses in general. You might want to show it on click too. (and hide it if it's clicked again).

Reply:
Yeah, let's have it trigger on focus events too please, Erik.

A lot of the time these will live on links, so let's not have it on click as a general rule, but we might definitely want to add the ability for that further down the line.

Reply:
A good way to _conditionally_ enable it on clicks would be to pass a data-attribute if it should. We can also do, semantically: `click` only if _not_ `a[href]:not([href=""])` and _not_ `button:not([action="tooltip"])`.

Would allow tooltips to trigger on links without `href` (anchors without destination), and on buttons with explicit action set.

Reply:
> but we might definitely want to add the ability for that further down the line.

Let's defer this to when we actually need it then.